core/src/main/java/club/mondaylunch/gatos/core/data/DataTypeRegistry.java

```diff
@@ -0,0 +1,31 @@
+package club.mondaylunch.gatos.core.data;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class DataTypeRegistry {
```

Comment:
I'm tempted to write a generic `Registry` class, since this is mostly the same as `NodeTypeRegistry`. Perhaps with a `.codec()` method to create a codec from the registry.